app/services/schools/register_ect.rb

```diff
@@ -68,6 +70,44 @@ def create_teacher!
       @teacher = ::Teacher.create_with(trs_first_name:, trs_last_name:, corrected_name:).find_or_create_by!(trn:)
     end
 
+    def registration_period
+      @registration_period ||= RegistrationPeriod.for_date(started_on)
+    end
+
+    def active_lead_provider
+      @active_lead_provider ||= ActiveLeadProvider.find_by(
+        lead_provider:,
+        registration_period:
+      ) || raise("Missing ActiveLeadProvider for #{lead_provider&.name} in #{registration_period&.year}")
+    end
+
+    def school_partnership
+      provider = active_lead_provider
+      return unless provider
+
+      SchoolPartnership
+        .joins(:lead_provider_delivery_partnership)
+        .find_by(
+          school:,
+          lead_provider_delivery_partnerships: {
+            active_lead_provider_id: provider.id
+          }
+        )
+    end
+
+    def expression_of_interest
+      school_partnership ? nil : active_lead_provider
+    end
+
+    def create_training_period!
+      @training_period = ::TrainingPeriod.create!(
+        ect_at_school_period:,
+        started_on: ect_at_school_period.started_on,
+        school_partnership:,
+        expression_of_interest:
+      )
+    end
```

Comment:
I wouldn't worry yet but maybe when we add the mentor version of this it might make sense to deal with training period creation in its own service class. There will be a few other things that will trigger their creation too.